dnf/transaction_sr.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import libdnf
+
+from dnf.i18n import _
+import dnf.exceptions
+
+import json
+
+
+VERSION_MAJOR = "1"
+VERSION_MINOR = "0"
+VERSION = "%s.%s" % (VERSION_MAJOR, VERSION_MINOR)
+
+
+class StoredTransactionVersionError(dnf.exceptions.Error):
+    def __init__(self, value=None):
+        super(StoredTransactionVersionError, self).__init__(value)
+
+
+def check_version(version, filename):
+    major, minor = version.split('.')
+    if major != VERSION_MAJOR:
+        raise StoredTransactionVersionError(
+            _('Incompatible major version {major} in "{fn}", '
+                'supported major version is {major_supp}.'
+            ).format(major=major, fn=filename, major_supp=VERSION_MAJOR)
+        )
+
+    if minor < VERSION_MINOR:
+        raise StoredTransactionVersionError(
+            _('Incompatible minor version {minor} in "{fn}", '
+                'supported minor version is {minor_supp} or greater.'
+            ).format(minor=minor, fn=filename, minor_supp=VERSION_MINOR)
+        )
+
+
+def store_transaction(transaction, filename="transaction.json"):
+    data = {
+        "version": VERSION,
+        "solver_flags": "",  # TODO
+    }
+    rpms = {}
+    groups = {}
+    environments = {}
+
+    for i in transaction.packages():
```

Comment:
In dnf we often use tsi as a transaction item. I suggest that `tsi` will be more informative then `i`.

Reply:
Even though I'm not sure how `tsi` became an abbreviation for `transaction` :slightly_smiling_face: (I'll use `tsi` according to your suggestion)